src/GenFu.Web/Models/SourceCode.cs

```diff
@@ -9,52 +9,45 @@
 using System.Linq;
 using System.Text;
 using System.Globalization;
-//using Microsoft.Dnx.Compilation;
-//using Microsoft.Dnx.Runtime;
-//using Microsoft.Dnx.Compilation.CSharp;
 using Microsoft.Extensions.PlatformAbstractions;
 using Microsoft.DotNet.ProjectModel.Compilation;
 using System.Runtime.Loader;
-using Microsoft.VisualStudio.Web.CodeGeneration.DotNet;
-//using Microsoft.VisualStudio.Web.CodeGeneration.DotNet;
+using GenFu.Web.Helpers;
 
 namespace GenFu.Web.Models
 {
-	public static class SourceCodeHelpers
-	{
-		public static readonly Type[] SupportedTypes = new Type[]
-		{
-			typeof(Object),
-			typeof(short),
-			typeof(float),
-			typeof(Int32),
-			typeof(Int16),
-			typeof(double),
-			typeof(decimal),
-			typeof(Guid),
-			typeof(long),
-			typeof(ulong),
-			typeof(uint),
-			typeof(ushort),
-			typeof(bool),
-			typeof(string),
-			typeof(DateTime)
-		};
-
-		public static MetadataReference GetMeta(this Type type)
+    public class SourceCode
+    {
+		public SourceCode()
 		{
-			return MetadataReference.CreateFromFile(type.GetTypeInfo().Assembly.Location);
+			//Check to see if we need to compute Metadata References
+			// build references up
+			if (references == null || references.Count == 0)
+			{
+				foreach (var type in SourceCodeHelpers.SupportedTypes)
+				{
+					references.Add(type.GetMeta());
+				}
+
+				//compare to see if we got duplicate data
+				var noDuplicatesList = references.GroupBy(x => x.Display).Select(x => x.First()).ToList();
+
+				references.Clear();
+
+				references.AddRange(noDuplicatesList);
+			}
+
+
 		}
-	}
 
-    public class SourceCode
-    {
 		// todo: make not hacky
 		public AssemblyLoadContext Accessor { get; set; }
 
-		public ILibraryExporter LibraryExporter { get; set; }
+		//cache meta data, these bits are costly to load and can cause frequent GC
+		//also since we have a hard coded list there's no need to recompute this per request
+		private static readonly List<MetadataReference> references = new List<MetadataReference>();
```

Comment:
One particular item to note in the reworking of the code compilation, I choose to cache MetadataReferences as static variable (list) as they’re expensive to load per request and don’t change between requests. Reloading the references per each request causes significant heap growth to the point where the GC is running per each request.